services/bots/src/discord/commands/components.ts

```diff
@@ -0,0 +1,128 @@
+import { TransformPipe } from '@discord-nestjs/common';
+import {
+  DiscordTransformedCommand,
+  Payload,
+  TransformedCommandExecutionContext,
+  Param,
+  UsePipes,
+  On,
+} from '@discord-nestjs/core';
+import { CommandHandler, DiscordCommandClass } from '../discord.decorator';
+import { AutocompleteInteraction, EmbedBuilder } from 'discord.js';
+import { reportException } from '@lib/sentry/reporting';
+import { ComponentDataService } from '../services/component-data';
+
+class ComponentsDto {
+  @Param({
+    name: 'component',
+    description: 'What is the name of the component?',
+    required: true,
+    autocomplete: true,
+  })
+  component: string;
+}
+
+@DiscordCommandClass({
+  name: 'component',
+  description: 'Returns information about an component',
+})
+@UsePipes(TransformPipe)
+export class ComponentsCommand implements DiscordTransformedCommand<ComponentsDto> {
```

Comment:
Since this is ESPHome specific, would it make sense to add that to the filename/class name ?

Reply:
Yes, no, I'm really not sure.
We do not do it for HA, so felt strange here.

Reply:
I think that it's something we should do. It will be a mess otherwise to distinguish what is where, except for that one file where it's mapped to org.

Reply:
Let's merge this as is, then I can do a split PR to segment it into 3 groups.
- Common
- HA
- ESPHome

Reply:
https://github.com/home-assistant/service-hub/pull/104